contracts/contracts/strategies/OUSDCurveAMOStrategy.sol

```diff
@@ -0,0 +1,689 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+/**
+ * @title Curve Automated Market Maker (AMO) Strategy
+ * @notice AMO strategy for the Curve OUSD/USDT pool
+ * @author Origin Protocol Inc
+ */
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import { IERC20, InitializableAbstractStrategy } from "../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { Tether } from "../interfaces/Tether.sol";
+import { ICurveStableSwapNG } from "../interfaces/ICurveStableSwapNG.sol";
+import { ICurveLiquidityGaugeV6 } from "../interfaces/ICurveLiquidityGaugeV6.sol";
+import { IBasicToken } from "../interfaces/IBasicToken.sol";
+import { ICurveMinter } from "../interfaces/ICurveMinter.sol";
+
+contract OUSDCurveAMOStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeCast for uint256;
+
+    /**
+     * @dev a threshold under which the contract no longer allows for the protocol to manually rebalance.
+     *      Guarding against a strategist / guardian being taken over and with multiple transactions
+     *      draining the protocol funds.
+     */
+    uint256 public constant SOLVENCY_THRESHOLD = 0.998 ether;
+
+    // New immutable variables that must be set in the constructor
+    /**
+     * @notice Address of the hard asset (weth, usdt, usdc).
+     */
+    Tether public immutable hardAsset;
```

Comment:
Can we use safeERC20 for `hardAsset`? We don't use the return value anywhere, but would still be good to stay consistent

Reply:
Good idea yes! Fixed in this commit: [1636c07](https://github.com/OriginProtocol/origin-dollar/pull/2436/commits/1636c07bc3c987b2c5aac8c801d1fd415d19049c)